src/analyses/constant_propagator.cpp

```diff
@@ -167,8 +253,11 @@ bool constant_propagator_domaint::two_way_propagate_rec(
 #ifdef DEBUG
   std::cout << "two_way_propagate_rec: " << from_expr(ns, "", expr) << '\n';
 #endif
-  bool change = false;
 
+  bool change=false;
+
+  // this seems to be buggy at present
```

Comment:
We got some wrong results on an industrial example which went away when two-way-propagation was disabled. I'll look into it more and try to provide a regressiont test and fix.